esp/src/src-react/components/Title.tsx

```diff
@@ -32,6 +34,14 @@ export const DevTitle: React.FunctionComponent<DevTitleProps> = ({
     const [currentUser, setCurrentUser] = React.useState(undefined);
     const [isOpen, { setTrue: openPanel, setFalse: dismissPanel }] = useBoolean(false);
 
+    const [showBanner, setShowBanner] = React.useState(false);
+    const [showBannerConfig, setShowBannerConfig] = React.useState(false);
+    const [bannerMessage, setBannerMessage] = React.useState("");
+    const [bannerColor, setBannerColor] = React.useState("");
+    const [bannerSize, setBannerSize] = React.useState("");
+
+    const activity = useConst(ESPActivity.Get());
+
```

Comment:
The encapsulation feels wrong here (the "activity" instance should be moved into the Banner file), I would suggest:
* Rename BannerConfig.tsx to Banner.tsx
* Create a new component "BannerMessageBar" inside that file
* Move the "activity" instance (its a global singleton anyway) into both the  BannerConfig and BannerMessageBar components